### PR TITLE
docs(architecture): proofread, updated Dart app

### PR DESCRIPTION
**Pull request by chalin**
Note that there are two commits: one for each of TS- and Dart-side changes.

- e2e tests now also cover the tax calculator.
- Dart app updated to match TS (it had no sales tax calculator).
- TS sample source cleanup (e.g. removed many unnecessary `docregions`).
- Prose updated to include @kwalrath's revisions from a while ago, with some of my edits as well.

Contributes to #1598 and #1508.

### CODE REVIEW

**Comment by chalin** [comment]
@kwalrath @thso : Dart-side commit ready for review.
@Foxandxss @wardbell : TS-side commit ready for review. As usual, it is best to review the [`.jade` file while ignoring changes in indentation](https://github.com/angular/angular.io/pull/1807/commits/5488fec1bf0fcce05e2a075f8bf1dbbe065733fa?w=1). Note that most of the TS Jade file edits were from Kathy's work from a while back.

File: public/docs/dart/latest/guide/architecture.jade [hunk 1]
```diff
@@ -1,451 +1,51 @@
...
-  * `selector`: A CSS selector that tells Angular to create and insert an instance of this component
-  where it finds a `<hero-list>` tag in *parent* HTML.
-  For example, if an app's  HTML contains `<hero-list></hero-list>`, then
-  Angular inserts an instance of the `HeroListComponent` view between those tags.
+block dart-terminology
+  .callout.is-helpful
+    header Dart difference: Module is a compilation unit
+    :marked
+      The term "module" used in this guide refers to a
+      Dart _compilation unit_, i.e., a _library_ or _part_.
```
Comment (kwalrath) [inline public/docs/dart/latest/guide/architecture.jade]: i.e. -> that is

(many people don't know what i.e. means)

Actually, though... do we really need to mention parts? They're strongly discouraged. Also, people might not know that each file without a `library` directive is itself a library. Perhaps instead:

In this guide, the term _module_ refers to a Dart _compilation unit_, such as a _library_. If a Dart file has no `library` or `part` directive, then that file itself is a library and thus a compilation unit. For more information...

File: public/docs/dart/latest/guide/architecture.jade [hunk 2]
```diff
@@ -1,451 +1,51 @@
...
+    Angular 2 for Dart is published as the `angular2` package, which
+    (like many other Dart packages) is available via the Pub tool.
 
-  * `selector`: A CSS selector that tells Angular to create and insert an instance of this component
-  where it finds a `<hero-list>` tag in *parent* HTML.
-  For example, if an app's  HTML contains `<hero-list></hero-list>`, then
-  Angular inserts an instance of the `HeroListComponent` view between those tags.
+block dart-terminology
+  .callout.is-helpful
+    header Dart difference: Module is a compilation unit
```
Comment (kwalrath) [inline public/docs/dart/latest/guide/architecture.jade]: Module -> A module
OR
Libraries are modules

File: public/docs/dart/latest/guide/architecture.jade [hunk 3]
```diff
@@ -1,451 +1,51 @@
...
-
-.l-sub-section
+block angular-library-modules
   :marked
-    While **a component is technically a directive**,
-    components are so distinctive and central to Angular applications that we chose
-    to separate components from directives in this architectural overview.
-:marked
-  Two *other* kinds of directives exist: _structural_ and _attribute_ directives.
+    Some modules are libraries.
```
Comment (kwalrath) [inline public/docs/dart/latest/guide/architecture.jade]: Just some?

File: public/docs/ts/latest/guide/architecture.jade [hunk 4]
```diff
@@ -1,190 +1,218 @@
...
 figure
   img(src="/resources/images/devguide/architecture/overview2.png" alt="overview" style="margin-left:-40px;" width="700")
+
 :marked
   The architecture diagram identifies the eight main building blocks of an Angular 2 application:
-  1. [Module](#module)   
-  1. [Component](#component)
-  1. [Template](#template)
+
+  1. [Modules](#modules)
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: I had left the Modules section out of the Dart version because (1) it was added late to the TS version (which is why we already had a handy, module-less image) and (2) it seemed like it wasn't really necessary, from a Dart perspective.

Are "modules" a concept used in other pages?

File: public/docs/ts/latest/guide/architecture.jade [hunk 5]
```diff
@@ -1,190 +1,218 @@
...
+  Perhaps the first module we meet is a module that exports a *component* class.
   The component is one of the basic Angular blocks, we write a lot of them,
   and we'll talk about components in the next segment. For the moment it is enough to know that a
   component class is the kind of thing we'd export from a module.
-  
-  Most applications have an `AppComponent`. By convention, we'll find it in a file named `app.component.ts`. 
-  Look inside such a file and we'll see an `export` statement like this one.
-+makeExample('architecture/ts/app/app.component.ts', 'export', 'app/app.component.ts (excerpt)')(format=".")
+
+  Most applications have an `AppComponent`. By convention, we'll find it in a file named `app.component.ts`.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: Fix filename.

File: public/docs/ts/latest/guide/architecture.jade [hunk 6]
```diff
@@ -1,190 +1,218 @@
...
 :marked
-  Angular apps are modular. 
-  
+  Angular apps are modular.
+
   In general we assemble our application from many **modules**.
-  
+
   A typical module is a cohesive block of code dedicated to a single purpose.
   A module **exports** something of value in that code, typically one thing such as a class.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: I'd leave the word "export" out of the Dart version, since in TS it refers to an explicit **export** keyword.

File: public/docs/ts/latest/guide/architecture.jade [hunk 7]
```diff
@@ -1,190 +1,218 @@
...
+       approach using that syntax. That's why we list *Module* among the basic building blocks.
+
+       Angular itself doesn't require a modular approach nor this particular syntax. Don't use it if you don't want it.
+       Each chapter has plenty to offer after you steer clear of the `import` and `export` statements.
+
+       Find setup and organization clues in the JavaScript track (select it from the combo-box at the top of this page)
+       which demonstrates Angular 2 development with plain old JavaScript and no module system.
+
+:marked
+  Perhaps the first module we meet is a module that exports a *component* class.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: again with the "export"

In Dart (if we need to keep this section), I'd use something like "defines" instead.

File: public/docs/ts/latest/guide/architecture.jade [hunk 8]
```diff
@@ -1,190 +1,218 @@
...
   The component is one of the basic Angular blocks, we write a lot of them,
   and we'll talk about components in the next segment. For the moment it is enough to know that a
   component class is the kind of thing we'd export from a module.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: "export" is again questionable

File: public/docs/ts/latest/guide/architecture.jade [hunk 9]
```diff
@@ -1,190 +1,218 @@
...
+    `AppComponent` class is public and accessible to other modules of the application.
+
 :marked
-  The `export` statement tells TypeScript that this is a module whose
-  `AppComponent` class is public and accessible to other modules of the application.
-  
   When we need a reference to the `AppComponent`, we **import** it like this:
-+makeExample('architecture/ts/app/main.ts', 'import', 'app/main.ts (excerpt)')(format=".")
+
++makeExcerpt('app/main.ts', 'import')
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: It seems like this section, if we need it at all, could be a lot smaller.

File: public/docs/ts/latest/guide/architecture.jade [hunk 10]
```diff
@@ -1,190 +1,218 @@
...
 :marked
-  Some modules are libraries of other modules.
-  
-  Angular itself ships as a collection of library modules within several npm packages.
-  Their names begin with the `@angular` prefix. 
-  Each Angular library contains a [barrel](../glossary.html#barrel) module
-  that is actually a public façade over several logically-related private modules.
-  
-  The `@angular/core` library is the primary Angular library module from which we get most of what we need.
+  `!{_at_angular}/core` is the primary Angular library from which we get most of what we need.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: This discussion of specific libraries seems like the most useful part of this section.

I'm just not sure whether the concept of "modules" is something we need for the Dart side.

File: public/docs/ts/latest/guide/architecture.jade [hunk 11]
```diff
@@ -1,190 +1,218 @@
...
+
+      The infrastructure *behind* module loading and importing is an important subject.
+      But it's a subject outside the scope of this introduction to Angular.
+      While we're focused on our application, *import* and *export*
+      is about all we need to know.
+
 :marked
   The key take-aways are:
   * Angular apps are composed of modules.
   * Modules export things &mdash; classes, function, values &mdash; that other modules import.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: "export" again

File: public/docs/ts/latest/guide/architecture.jade [hunk 12]
```diff
@@ -1,190 +1,218 @@
...
   * We prefer to write our application as a collection of modules, each module exporting one thing.
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: exporting -> providing?

File: public/docs/ts/latest/guide/architecture.jade (hunk 4 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Yes, "module" is used throughout the docs. The concept of a “module” is ​*central*​ to Angular, and to the approach to app development that we want to promote.

File: public/docs/dart/latest/guide/architecture.jade (hunk 1 above)
Comment (chalin) [inline public/docs/dart/latest/guide/architecture.jade]: Fixed as you suggested with tweaks we agree to elsewhere (in particular clarifying that a Dart package is also a module).

File: public/docs/dart/latest/guide/architecture.jade (hunk 2 above)
Comment (chalin) [inline public/docs/dart/latest/guide/architecture.jade]: Changed to
> Dart difference: Modules are compilation units or packages

File: public/docs/dart/latest/guide/architecture.jade (hunk 3 above)
Comment (chalin) [inline public/docs/dart/latest/guide/architecture.jade]: I was trying to mimic the TS prose. I just dropped that sentence since it isn't very helpful.

File: public/docs/ts/latest/guide/architecture.jade (hunk 5 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Oh, I forgot to come back to that. Thanks for catching it! Fixed.

File: public/docs/ts/latest/guide/architecture.jade [hunk 13]
```diff
@@ -193,196 +221,214 @@ figure
...
-  
-  **Attribute** directives alter the appearance or behavior of an existing element. 
-  In templates they look like regular HTML attributes, hence the name.
-  
-  The `ngModel` directive, which implements two-way data binding, is an example of an attribute directive.
-+makeExample('architecture/ts/app/hero-detail.component.html', 'ngModel')(format=".")
+
+.l-sub-section
+  :marked
+    In Dart, **the only value that is true is the boolean value `true`**; all
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: Perhaps this should be a DART DIFFERENCE in the Dart version only? I can't think of anywhere else where we have Dart info in the TS docs.

File: public/docs/ts/latest/guide/architecture.jade [hunk 14]
```diff
@@ -193,196 +221,214 @@ figure
...
+  In templates they look like regular HTML attributes, hence the name.
+
+  The `ngModel` directive, which implements two-way data binding, is
+  an example of an attribute directive. `ngModel` modifies the behavior of
+  an existing element (typically an `<input>`)
   by setting its display value property and responding to change events.
-  
-  Angular ships with a few other directives that either alter the layout structure 
+
++makeExcerpt('app/hero-detail.component.html', 'ngModel')
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: I just realized that a lot of code snippets have headings that didn't used to. Do we really need the file info? I'd rather just concentrate on the code, for small snippets like this where the filename seems irrelevant.

File: public/docs/ts/latest/guide/architecture.jade [hunk 15]
```diff
@@ -193,196 +221,214 @@ figure
...
+:marked
+  Angular ships with a small number of other directives that either alter the layout structure
   (e.g. [ngSwitch](template-syntax.html#ngSwitch))
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: Use "for example" instead of "e.g.". From our internal style guide (which I hope will be available externally at some point):

**e.g.** Don't use. Instead, use phrases like "for example" or "for instance." Too many people mix up "e.g." and "i.e."

File: public/docs/ts/latest/guide/architecture.jade [hunk 16]
```diff
@@ -392,151 +438,160 @@ figure
...
   Angular uses dependency injection to provide new components with the services they need.
 <br clear="all">
 :marked
-  In TypeScript, Angular can tell which services a component needs by looking at the types of its constructor parameters.
-  For example, the constructor of our `HeroListComponent` needs the `HeroService`:
-+makeExample('architecture/ts/app/hero-list.component.ts', 'ctor', 'app/hero-list.component (constructor)')(format=".")
+  Angular can tell which services a component needs by looking at the types of its constructor parameters.
+  For example, the constructor of our `HeroListComponent` needs a `HeroService`:
+
++makeExcerpt('app/hero-list.component.ts (constructor)', 'ctor')
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: It seems like this transcript should include the place where _heroService is declared, so you can see it's a HeroService.

File: public/docs/ts/latest/guide/architecture.jade [hunk 17]
```diff
@@ -1,190 +1,218 @@
...
-  A `HeroListComponent`, for example, might have a `heroes` property that returns an array of heroes
+  A `HeroListComponent`, for example, might have a `heroes` property that returns an !{_array} of heroes
   that it acquired from a service.
-  It might have a `selectHero()` method that sets a `selectedHero` property when the user clicks on a hero from that list.
-  It might be a class like this:
+  It might have a `selectHero()` method that sets a `selectedHero` property when the user clicks to choose a hero from that list.
+  The component might be a class like this:
 
-+makeExample('architecture/ts/app/hero-list.component.ts', 'class', 'app/hero-list.component.ts')
++makeExcerpt('app/hero-list.component.ts', 'class')
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: This excerpt uses a couple of APIs that aren't explained here (or anywhere on this page): OnInit, ngOnInit.

If we can't hide them, let's add an aside that briefly explains them and/or refers to where they're explained.

File: public/docs/ts/latest/guide/architecture.jade [hunk 18]
```diff
@@ -392,151 +438,160 @@ figure
...
-+makeExample('architecture/ts/app/main.ts', 'bootstrap','app/main.ts (excerpt)')(format=".")
+  
++makeExcerpt('app/main.ts', 'bootstrap')
+
 :marked
-  Alternatively, we might register at a component level ...
-+makeExample('architecture/ts/app/hero-list.component.ts', 'providers','app/hero-list.component.ts (excerpt)')(format=".")
+  Alternatively, we might register at a component level:
+
++makeExcerpt('app/hero-list.component.ts', 'providers')
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: This snippet is uglier than it used to be, due to the `/* . . . */`s and `implements OnInit`. Any way to make it cleaner?

File: public/docs/ts/latest/guide/architecture.jade [hunk 19]
```diff
@@ -392,151 +438,160 @@ figure
...
-  >**Change Detection** - Learn how Angular decides that a component property value has changed and 
-  when to update the screen. 
-  Learn how it uses **zones** to intercept asynchronous activity and run its change detection strategies. 
-  
-  >**[Component Router](router.html)** - With the Component Router service, users can navigate a multi-screen application 
+
+  Here is a brief, alphabetical list of other important Angular features and services.
+  Most of them are covered in this Developers Guide (or soon will be).
+
+  > [**Animations**](animations.html): The animation library makes it easy for developers to animate component behavior
```
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: This link doesn't work in the Dart version. Add a placeholder?

**Comment by kwalrath** [comment]
Looking good! Once you address a few comments, the Dart side is good to go.

File: public/docs/ts/latest/guide/architecture.jade (hunk 6 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: As agreed, I kept this first occurrence of 'export'.

File: public/docs/ts/latest/guide/architecture.jade (hunk 7 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Changed to "declare".

File: public/docs/ts/latest/guide/architecture.jade (hunk 13 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Right, this was meant to be only in the Dart version. Fixed.

File: public/docs/ts/latest/guide/architecture.jade (hunk 14 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Yes. I found that having, at this early stage, the context for the snippets helps create a better mental picture of how the pieces fit together (and where to find them in the sample sources without having to resort to a global text search). That's my 2 cents. If you prefer all of the snippets to be void of titles, that will be easy to fix.

File: public/docs/ts/latest/guide/architecture.jade (hunk 15 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: fixed

File: public/docs/ts/latest/guide/architecture.jade (hunk 16 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Good point. Fixed.

File: public/docs/ts/latest/guide/architecture.jade (hunk 17 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Lifecycle hooks chapter is mentioned in the paragraph that follows. I've added an explicit mention of `ngOnInit`.

File: public/docs/ts/latest/guide/architecture.jade (hunk 18 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Its uglier but provides context. The /*...*/ are easily removed. It's your call. (I'd leave the OnInit).

**Comment by chalin** [comment]
Dart-side issues addressed.

**Comment by kwalrath** [comment]
LGTM! @Foxandxss or @wardbell, who's going to review the TS side?

File: public/docs/_examples/architecture/dart/lib/sales_tax_component.dart [hunk 20]
```diff
@@ -0,0 +1,28 @@
+import 'package:angular2/core.dart';
+
+import 'sales_tax_service.dart';
+import 'tax_rate_service.dart';
+
+@Component(
+    selector: 'sales-tax',
+    template: '''
+      <h2>Sales Tax Calculator</h2>
+      Amount: <input #amountBox (input)="0" type="number">
```
Comment (Foxandxss) [inline public/docs/_examples/architecture/dart/lib/sales_tax_component.dart]: What triggered this change (also in TS)? I am not sure I like `(input)` as the event name. I don't know, I guess it is just me but I liked `(change)` more. Input is also the element name so as an event again is weird. Also `input` sounds more like an input (pun intended) than as an output.

On the other hand, the number input is really broken in other browsers. I don't think they are fixed yet.

Comment (chalin) [inline public/docs/_examples/architecture/dart/lib/sales_tax_component.dart]: Triggering on `onInput` rather than `onChange` makes for a more responsive user experience (UX). Again, using `type="number"` makes for a better UX in those browsers that support it, and makes no difference (from the app processing point of view) in those browsers that do not support it.

**Comment by chalin** [comment]
@Foxandxss : please see my response to your line comment. Otherwise, are we good to go on this one?

File: public/docs/ts/latest/guide/architecture.jade [hunk 21]
```diff
@@ -73,18 +73,19 @@ block modules-are-optional
...
+- var _app_comp_filename = _docsFor == 'dart' ? 'app_component.dart' : 'app.component.ts';
 :marked
   Perhaps the first module we meet is a module that exports a *component* class.
   The component is one of the basic Angular blocks, we write a lot of them,
   and we'll talk about components in the next segment. For the moment it is enough to know that a
   component class is the kind of thing we'd export from a module.
 
-  Most applications have an `AppComponent`. By convention, we'll find it in a file named `app.component.ts`.
+  Most applications have an `AppComponent`. By convention, we'll find it in a file named `!{_app_comp_filename}`.
   Look inside such a file and we'll see a declaration like this one.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: Replace like with such as:
Look inside such a file and we'll see a declaration such as this one.

File: public/docs/_examples/architecture/dart/lib/sales_tax_component.dart (hunk 20 above)
Comment (wardbell) [inline public/docs/_examples/architecture/dart/lib/sales_tax_component.dart]: Please revert to `(change)`.  We're not trying to be responsive. We're trying to make a point about how `(someEvent)="0"` is a trick to cause dirty checking when the user enters something.  `(input)` invites the kind of distracting question that Jesus asked.

Please revert `type="number"`. This is a distraction from the point. Again, we're not looking to produce the best app solution here. We're trying to make a narrow point. Let's not add distractions from that point.

**Comment by wardbell** [comment]
Please update ts/architecture.jade as follows:
line 84: "Look inside such a file and we'll see a declaration such as this one."
line 384: "They tend to appear within an element tag as attributes do,"

**Comment by wardbell** [comment]
I am uncomfortable with certain changes in the e2e tests, most especially the tests-that-are-not-tests such as 
```
  it(`selects ${targetHero.name} from hero list`, function () {
    let hero = element(by.cssContainingText('li', targetHero.name));
    hero.click();
    // Nothing specific to expect other than lack of exceptions.
  });
```
and
```
  it(`can update hero name`, () => {
    addToHeroName(nameSuffix);
    // Nothing specific to expect other than lack of exceptions.
  });
```
The second is also not accepted because it delegates to `addToHeroName`, a helper function that is used exactly ONCE, in this place only. That move detracts from readability and maintenance w/o compensating benefits.

I think I know why you created these tests; they are a clever way to slip in an async operation w/o waiting for it. But they depend upon test order which test suites should not do. Move one of these tests and the suite fails.

I realize that a great number of our e2e tests have this failing (when you see `beforeAll` loading the browser just once, you can guess that cross-test pollution is a likely risk). I accept that to make these tests faster. But I don't accept splitting what is actually ONE test over two tests in order to avoid the async setup operation.  Please think of a better way.

**Comment by wardbell** [comment]
Unfortunately, this PR is NOT based on an IdeaBlade branch. That means I can't fix it.

You and Kathy have the ability to add branches to `ideablade/angular.io`.

Please base all future PRs on branches in  `ideablade/angular.io` or at least all future branches that involve this many changes.

File: public/docs/_examples/architecture/ts/app/hero-list.component.html [hunk 22]
```diff
@@ -2,8 +2,10 @@
 <h2>Hero List</h2>
 
 <p><i>Pick a hero from the list</i></p>
-<div *ngFor="let hero of heroes" (click)="selectHero(hero)">
-  {{hero.name}}
-</div>
+<ul>
```
Comment (wardbell) [inline public/docs/_examples/architecture/ts/app/hero-list.component.html]: Why switch from `<div>` to `<ul><li>`?  That a distraction. I wanted _less code on the page_. I didn't want to have to deal with the bullets of a `<ul>`.

I'm not going to insist on reverting this.  But I think you are mistakenly trying to improve the code at the expense of the clarity of the Template Syntax. 

"LESS HTML IS MORE"

File: public/docs/_examples/architecture/ts/app/sales-tax.component.ts [hunk 23]
```diff
@@ -1,42 +1,25 @@
...
 
-// #docregion metadata
-// #docregion providers
 @Component({
-// #enddocregion providers
   selector:    'sales-tax',
   template: `
     <h2>Sales Tax Calculator</h2>
-    Amount: <input #amountBox (change)="0">
+    Amount: <input #amountBox (input)="0" type="number">
```
Comment (wardbell) [inline public/docs/_examples/architecture/ts/app/sales-tax.component.ts]: Please revert for reasons given in the Dart version.

File: public/docs/_examples/architecture/ts/app/sales-tax.service.ts [hunk 24]
```diff
@@ -1,20 +1,14 @@
...
   constructor(private rateService: TaxRateService) { }
+
   getVAT(value: string | number) {
-    let amount: number;
-    if (typeof value === 'string') {
-      amount = parseFloat(value);
-    } else {
-      amount = value;
-    }
+    let amount = (typeof value === 'string') ?
```
Comment (wardbell) [inline public/docs/_examples/architecture/ts/app/sales-tax.service.ts]: This is I like

File: public/docs/ts/latest/guide/architecture.jade (hunk 4 above)
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: We are about to have TWO notions of module when ***Application Modules*** land in RC5.  That will force the next revision of this doc.

File: public/docs/ts/latest/guide/architecture.jade (hunk 9 above)
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: What did you have in mind, KW?

File: public/docs/ts/latest/guide/architecture.jade [hunk 25]
```diff
@@ -193,196 +224,208 @@ figure
...
-  
+
   In fact, it really is *just a class*. It's not a component until we *tell Angular about it*.
-  
+
   We tell Angular that `HeroListComponent` is a component by attaching **metadata** to the class.
-  
-  The easy way to attach metadata in TypeScript is with a **decorator**. 
+
+  In !{_Lang}, we attach metadata by using an **!{_decorator}**.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: Is this going to render as "an decorator" in TS? I thought you had another token that distinguished "a decorator" (ts) from "an annotation" (dart).

File: public/docs/ts/latest/guide/architecture.jade [hunk 26]
```diff
@@ -193,196 +224,208 @@ figure
...
+
+:marked
+  - `selector`: CSS selector that tells Angular to create and insert an instance of this component
+  where it finds a `<hero-list>` tag in *parent* HTML.
+  For example, if an app's  HTML contains `<hero-list></hero-list>`, then
+  Angular inserts an instance of the `HeroListComponent` view between those tags.
+
+  - `templateUrl`: address of this component's template, which we showed [above](#templates).
+
+  - `directives`: !{_array} of the components or directives that *this* template requires.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: For future reference, I don't mind using the word "list" in TypeScript instead of "array" if that makes it easier for text sharing with Dart.

File: public/docs/ts/latest/guide/architecture.jade [hunk 27]
```diff
@@ -193,196 +224,208 @@ figure
...
+
 figure
   img(src="/resources/images/devguide/architecture/template-metadata-component.png" alt="Metadata" align="left" style="height:200px; margin-left:-40px;margin-right:10px" )
+
 :marked
-  The `@Component` function takes the configuration object and turns it into metadata that it attaches
-  to the component class definition. Angular discovers this metadata at runtime and thus knows how to do "the right thing".
-  
-  The template, metadata, and component together describe the view.
+  At runtime, Angular discovers the metadata specified by the `@Component`
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: Can't say "at runtime" because the Angular Static Compiler will do it "offline". Maybe we just say "Angular discovers ..."

File: public/docs/ts/latest/guide/architecture.jade [hunk 28]
```diff
@@ -193,196 +224,208 @@ figure
...
+  annotation. That's how Angular learns how to do "the right thing".
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: That's how Angular learns to do "the right thing".

File: public/docs/ts/latest/guide/architecture.jade [hunk 29]
```diff
@@ -193,196 +224,208 @@ figure
...
+
+  The template, metadata, and component together describe a view.
 
-  We apply other metadata decorators in a similar fashion to guide Angular behavior. 
-  The `@Injectable`, `@Input`, `@Output`, `@RouterConfig` are a few of the more popular decorators
+  We apply other metadata !{_decorator}s in a similar fashion to guide Angular behavior.
+  `@Injectable`, `@Input`, `@Output`, and `@RouterConfig` are a few of the more popular !{_decorator}s
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: `@RouterConfig` is dead as of the new router. Let's just say "`@Injectable`, `@Input`, and `@Output`"

File: public/docs/ts/latest/guide/architecture.jade [hunk 30]
```diff
@@ -193,196 +224,208 @@ figure
...
+
++makeExcerpt('app/hero-list.component.1.html', 'binding')
+
 :marked
-  * The `{{hero.name}}` [*interpolation*](displaying-data.html#interpolation) 
-  displays the component's `hero.name` property value within the `<div>` tags.
-  
-  * The `[hero]` [*property binding*](template-syntax.html#property-binding) passes the `selectedHero` from
+  * The `{{hero.name}}` [*interpolation*](displaying-data.html#interpolation)
+  displays the component's `hero.name` property value within the `<li>` tags.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: See my note about this in the code sample. I'm not insisting on reverting this one. But I want to register my distress with the unnecessary HTML bloat introduced by switching from `<div>` to  `<ul><li>`.  

Reasonable minds might differ I suppose.

File: public/docs/ts/latest/guide/architecture.jade [hunk 31]
```diff
@@ -193,196 +224,208 @@ figure
...
 :marked
   In two-way binding, a data property value flows to the input box from the component as with property binding.
   The user's changes also flow back to the component, resetting the property to the latest value,
   as with event binding.
-  
-  Angular processes *all* data bindings once per JavaScript event cycle, 
-  depth-first from the root of the application component tree. 
+
+  Angular processes *all* data bindings once per JavaScript event cycle,
+  depth-first from the root of the application component tree.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: Actually, this is wrong. Angular processes bindings from the root of the app down.  

I don't think we need to specify the details here. Let's just say something like:

>Angular processes *all* data bindings once per JavaScript event cycle from the root of the application component tree down through the branches to the leaves.

File: public/docs/ts/latest/guide/architecture.jade [hunk 32]
```diff
@@ -193,196 +224,208 @@ figure
...
 :marked
-  ## The Service
+  ## Services
 figure
   img(src="/resources/images/devguide/architecture/service.png" alt="Service" style="float:left; margin-left:-40px;margin-right:10px" )
 :marked
-  "Service" is a broad category encompassing any value, function or feature that our application needs.
-  
-  Almost anything can be a service. 
+  _Services_ is a broad category encompassing any value, function, or feature that our application needs.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: _Service_ is a ...

I don't mind that it is not plural like the section header.

File: public/docs/ts/latest/guide/architecture.jade [hunk 33]
```diff
@@ -392,151 +435,160 @@ figure
...
 .l-main-section
-<a id="dependency-injection"></a>
 :marked
-  ## Dependency Injection
+  ## Dependency injection
 figure
   img(src="/resources/images/devguide/architecture/dependency-injection.png" alt="Service" style="float:left; width:200px; margin-left:-40px;margin-right:10px" )
 :marked
-  "Dependency Injection" is a way to supply a new instance of a class 
+  Dependency injection is a way to supply a new instance of a class
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: _Dependency injection_ (italics please)

File: public/docs/ts/latest/guide/architecture.jade (hunk 18 above)
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: I think we're losing the point by including to much code.

Why not say "Alternatively, we might register at a component level in the `providers` property of the component metadata:"

Then the only thing in the "providers region" of the sample is:
```
  // #docregion providers
  providers:   [HeroService]
  // #enddocregion providers
```
```

File: public/docs/ts/latest/guide/architecture.jade [hunk 34]
```diff
@@ -392,151 +435,160 @@ figure
...
+code-example().
   price | currency:'USD':true
 </div>
 :marked
-  >displays a price of "42.33" as `$42.33`.
-  
-  >**[Testing](testing.html)** - Angular provides a testing library for "unit testing" our application parts as they
-  interact with the Angular framework.
+  > It displays a price of "42.33" as `$42.33`.
+
```
Comment (wardbell) [inline public/docs/ts/latest/guide/architecture.jade]: We should add something about the Router. Maybe the following?
```
[**Router**](router.html): Navigate from page to page within the client application and never leave the browser.
```

File: public/docs/ts/latest/guide/architecture.jade (hunk 18 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: wfm
> Alternatively, we might register at a component level, in the providers property of the `@Component` metadata:

File: public/docs/_examples/architecture/dart/lib/sales_tax_component.dart (hunk 20 above)
Comment (chalin) [inline public/docs/_examples/architecture/dart/lib/sales_tax_component.dart]: Ok. Undone.

File: public/docs/_examples/architecture/ts/app/sales-tax.component.ts (hunk 23 above)
Comment (chalin) [inline public/docs/_examples/architecture/ts/app/sales-tax.component.ts]: Done.

File: public/docs/_examples/architecture/ts/app/hero-list.component.html (hunk 22 above)
Comment (chalin) [inline public/docs/_examples/architecture/ts/app/hero-list.component.html]: Ok.

File: public/docs/ts/latest/guide/architecture.jade (hunk 25 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Good catch. Fixed by writing: `!{_a} **!{_decorator}**`.

File: public/docs/ts/latest/guide/architecture.jade (hunk 26 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Ok.

File: public/docs/ts/latest/guide/architecture.jade (hunk 27 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Ok, but changed "discovers" to "reads" (otherwise the phrase doesn't really make sense).

File: public/docs/ts/latest/guide/architecture.jade (hunk 28 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: fixed

File: public/docs/ts/latest/guide/architecture.jade (hunk 29 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: done

File: public/docs/ts/latest/guide/architecture.jade (hunk 30 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Yes, noted.

File: public/docs/ts/latest/guide/architecture.jade (hunk 31 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: fixed

File: public/docs/ts/latest/guide/architecture.jade (hunk 32 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: ok, changed

File: public/docs/ts/latest/guide/architecture.jade (hunk 33 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: done

File: public/docs/ts/latest/guide/architecture.jade (hunk 34 above)
Comment (chalin) [inline public/docs/ts/latest/guide/architecture.jade]: Added.

**Comment by chalin** [comment]
@wardbell : thanks for the feedback. I believe that all of your concerns have been addressed in the last commit (https://github.com/angular/angular.io/pull/1807/commits/f38496f1a11831da67d772bb7ab2d1766830fe25). Let me know if there is anything else.

File: public/docs/ts/latest/guide/architecture.jade (hunk 9 above)
Comment (kwalrath) [inline public/docs/ts/latest/guide/architecture.jade]: It just seemed verbose, but fixing that can wait until the editors take a look.

**Comment by wardbell** [comment]
Thanks. Merging.
